src/adapter.ts

```diff
@@ -12,155 +12,184 @@
-        }
-        if (rule.length > 5) {
-            line.v5 = rule[5];
-        }
+  private savePolicyLine(ptype: string, rules: string[]): CasbinRuleAttributes {
+    const line: any = {p_type: ptype};
 
-        return line;
+    if (rules.length > 0) {
+      line.v0 = rules[0];
+    }
+    if (rules.length > 1) {
+      line.v1 = rules[1];
+    }
+    if (rules.length > 2) {
+      line.v2 = rules[2];
+    }
+    if (rules.length > 3) {
+      line.v3 = rules[3];
+    }
+    if (rules.length > 4) {
+      line.v4 = rules[4];
+    }
+    if (rules.length > 5) {
+      line.v5 = rules[5];
     }
 
-    /**
-     * savePolicy saves all policy rules to the storage.
-     */
-    public savePolicy(model: Model) {
-        this.dropTable();
-        this.createTable();
+    return line;
+  }
+
+  // savePolicy saves policy to database.
+  public async savePolicy(model: Model): Promise<boolean> {
+    // this.dropTable()
+    // this.createTable()
+    const {casbinRule} = this;
+    const lines: CasbinRuleAttributes[] = [];
+    let astMap = model.model.get('p');
+    // @ts-ignore
+    for (const [ptype, ast] of astMap) {
+      for (const rule of ast.policy) {
+        const line: CasbinRuleAttributes = this.savePolicyLine(ptype, rule);
+        lines.push(line);
+      }
+    }
 
-        const lines = [];
-        let astMap = model.get('p');
-        for (const [ptype, ast] of astMap) {
-            for (const rule of ast.Policy) {
-                const line = this.savePolicyLine(ptype, rule);
-                lines.push(line);
-            }
-        }
+    astMap = model.model.get('g');
+    // @ts-ignore
+    for (const [ptype, ast] of astMap) {
+      for (const rule of ast.policy) {
+        const line: CasbinRuleAttributes = this.savePolicyLine(ptype, rule);
+        lines.push(line);
+      }
+    }
 
-        astMap = model.get('g');
-        for (const [ptype, ast] of astMap) {
-            for (const rule of ast.Policy) {
-                const line = this.savePolicyLine(ptype, rule);
-                lines.push(line);
-            }
-        }
-
-        CasbinRule.bulkCreate(lines);
-    }
-
-    /**
-     * addPolicy adds a policy rule to the storage.
-     */
-    public addPolicy(sec: string, ptype: string, rule: string[]) {
-        throw new Error('not implemented');
-    }
-
-    /**
-     * removePolicy removes a policy rule from the storage.
-     */
-    public removePolicy(sec: string, ptype: string, rule: string[]) {
-        throw new Error('not implemented');
-    }
-
-    /**
-     * removeFilteredPolicy removes policy rules that match the filter from the storage.
-     */
-    public removeFilteredPolicy(sec: string, ptype: string, rule: string[]) {
-        throw new Error('not implemented');
+    await casbinRule.bulkCreate(lines);
+    return true;
+  }
+
+  // addPolicy adds a policy rule to the storage.
+  public async addPolicy(sec: string, ptype: string, rules: string[]): Promise<void> {
```

Comment:
Please do not implement ``addPolicy()`` and following functions. Let's make it work first.

Reply:
I think this function has no problem.

Reply:
You just copied your original code here. This makes this new repo not that meaningful if it's just a copy of the old repo. You can notice that the comment is also not the same.

Reply:
Comment or remove?

Reply:
There are too many other things we need to do first. I still suggest we do not care the optional functions like ``addPolicy()`` for now.

You can keep the code by yourself for now and when others things are done, you can submit it.

Reply:
Just leave it as before the PR.